### PR TITLE
Add support for pod5 files generated via dorado basecaller

### PR DESCRIPTION
**Pull request by Steffengreiner**
**What was changed**
Apparently the new Nanopore machines generate pod5 files instead of fast5 files dependent on the employed basecaller. Since the partner labs are updating their machines it's necessary to account for the newly valid minimal filetree structure

### CODE REVIEW

File: src/main/groovy/life/qbic/datamodel/instruments/OxfordNanoporeInstrumentOutputDoradoMinimal.groovy [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package life.qbic.datamodel.instruments
+
+
+/**
+ * Represents the Nanopore instrument output data structure schema.
```
Comment (wow-such-code) [inline src/main/groovy/life/qbic/datamodel/instruments/OxfordNanoporeInstrumentOutputDoradoMinimal.groovy:5]: I would mention Dorado here

File: src/main/groovy/life/qbic/datamodel/datasets/datastructure/folders/nanopore/Pod5SkipFolder.groovy [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package life.qbic.datamodel.datasets.datastructure.folders.nanopore
+
+import life.qbic.datamodel.datasets.datastructure.files.nanopore.Pod5File
+import life.qbic.datamodel.datasets.datastructure.folders.DataFolder
+
+/**
+ * A special case of a DataFolder, its name is always pod5_skip.
+ *
+ * Its children field contains either a list of type List<Pod5Files>
```
Comment (wow-such-code) [inline src/main/groovy/life/qbic/datamodel/datasets/datastructure/folders/nanopore/Pod5SkipFolder.groovy:9]: the either makes no sense - can it contain (barcoded?) folders?

File: src/main/groovy/life/qbic/datamodel/datasets/datastructure/folders/nanopore/Fast5SkipFolder.groovy [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package life.qbic.datamodel.datasets.datastructure.folders.nanopore
+
+import life.qbic.datamodel.datasets.datastructure.files.nanopore.Fast5File
+import life.qbic.datamodel.datasets.datastructure.folders.DataFolder
+
+/**
+ * A special case of a DataFolder, its name is always fast5_skip.
+ *
+ * Its children field contains either a list of type List<Fast5Files>
```
Comment (wow-such-code) [inline src/main/groovy/life/qbic/datamodel/datasets/datastructure/folders/nanopore/Fast5SkipFolder.groovy:9]: same problem as with pod5 skip

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 4]
```diff
@@ -101,24 +113,38 @@ final class OxfordNanoporeMeasurement {
...
-        }
+    // Condition one: Don't allow empty Fast5 pass and fail folder
+    private boolean areFast5FoldersInMeasurement() {
+        return isDataFolderInMeasurement("fast5pass") || isDataFolderInMeasurement("fast5fail")
+    }
+    // Condition two: Don't allow empty Fastq pass and fail folder
+    private boolean areFastQFoldersInMeasurement() {
+        return isDataFolderInMeasurement("fastqpass") || isDataFolderInMeasurement("fastqfail")
+    }
+    // Condition three: Don't allow empty Pod5 skip and fast5 skip folder
```
Comment (wow-such-code) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:138]: all others may be empty?

**Review by wow-such-code — CHANGES_REQUESTED** [review]
a few questions/nitpicks, but seems sensible so far!

File: src/main/groovy/life/qbic/datamodel/datasets/datastructure/folders/nanopore/Pod5SkipFolder.groovy (hunk 2 above)
Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/datasets/datastructure/folders/nanopore/Pod5SkipFolder.groovy:9]: Nice Catch it's a copy paste error from my side 😓

Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/datasets/datastructure/folders/nanopore/Pod5SkipFolder.groovy:9]: AFAIK the skip folders only contain pod5 files 
Bildschirmfoto 2023-11-03 um 15 22 55

File: src/main/groovy/life/qbic/datamodel/datasets/datastructure/folders/nanopore/Fast5SkipFolder.groovy (hunk 3 above)
Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/datasets/datastructure/folders/nanopore/Fast5SkipFolder.groovy:9]: Good eyes 👁️ (See my answer to the pod5 skip)

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy (hunk 4 above)
Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:138]: Yes the new data model foresees the following minimal and complete models 
**Minimal Model:** 
Bildschirmfoto 2023-11-03 um 15 22 55

**Complete Model:**
Bildschirmfoto 2023-11-03 um 15 22 59

**Review by wow-such-code — APPROVED** [review]

**Pull request merged**